client/src/unifyfs-fixed.c

```diff
@@ -471,6 +482,146 @@ static int unifyfs_split_index(
     return UNIFYFS_SUCCESS;
 }
 
+/*
+ * Clear all entries in the log index.  This only clears the metadata,
+ * not the data itself.
+ */
+static void
+unifyfs_clear_index(void)
+{
+    off_t num_entries = *unifyfs_indices.ptr_num_entries;
+    size_t index_size = sizeof(*unifyfs_indices.index_entry) * num_entries;
+
+    memset(unifyfs_indices.index_entry, 0, index_size);
+    *unifyfs_indices.ptr_num_entries = 0;
+}
+
+/*
+ * Sync all the extents to the server.  Clears the metadata index afterwards.
+ *
+ * Returns 0 on success, nonzero otherwise.
+ */
+int
+unifyfs_sync(int gfid)
+{
+    int ret;
+
+    if (unifyfs_flatten_writes) {
+        unifyfs_rewrite_index_from_seg_tree();
+    }
+
+    ret = invoke_client_fsync_rpc(gfid);
```

Comment:
Let's declare int ret here, and save a couple lines above.

Reply:
Ok